docs/development/maintainers.md

```diff
@@ -60,97 +124,178 @@ the latest release branch named `stable` (due to ReadTheDocs constraints).
 
 6. (Major release only). Create a new `stable` branch from this tag,
 
-   ```bash
-   git checkout -b stable
+   ```sh
+   git checkout main
+   git checkout -B stable
    git push upstream stable --force
    ```
 
-7. Revert the release commit. If making a major release, increment the version
-   to the next development version specified by Semantic Versioning.
-
+7. (Major or alpha but not minor release.) Set the version number back to the
+   development version. If you just released `0.22.0`, set the version to
+   `0.23.0.dev0`. If you just released `0.22.0a1` then you'll set the version to
+   `0.22.0.dev0`. Make a new commit from this and push it to upstream.
    ```sh
-   # If you just released 0.22.0, then set the next version to 0.23.0
+   git checkout main
    ./tools/bump_version.py --new-version 0.23.0.dev0
+   git add -u
+   git commit -m "0.23.0.dev0"
+   git push upstream main
    ```
 
-8. Update these instructions with any relevant changes.
+## Fixing documentation for a released version
 
-### Making a minor release
+Cherry pick the corresponding documentation commits to the `stable` branch. Use
+`git commit --amend` to add `[skip ci]` to the commit message.
 
-For a minor release, commits need to be added to the `stable` branch, ideally via a PR.
-This can be done with either,
+## Updating the Docker image
```

Comment:
I also use manually running GHA in UI (https://github.com/pyodide/pyodide/actions/workflows/docker_image.yml), as it does not require an access to `pyodide` org in docker hub. So probably it is worth to mention it too?